src/widgets/command_prompt.rs

```diff
@@ -0,0 +1,65 @@
+use termion::event::{Event, Key};
+use std::io::Write;
+use std::io::Error;
+
+use termion::clear::CurrentLine as ClearLine;
+use termion::cursor::Goto;
+use core::Command;
+
+/// Command prompt for xi-term.
+/// currently this is heavily inspired by vim
+/// and is just disigned to get a simple base to work off of.
+#[derive(Debug, Default)]
+pub struct CommandPrompt {
+    keys: String
```

Comment:
bikeshedding: maybe name if `chars`? Once written in on the screen, the command is more a sequence of characters than a sequence of key strokes.

Reply:
yeah thats a good point